data/src/file_upload_session.rs

```diff
@@ -36,6 +37,11 @@ use crate::remote_client_interface::create_remote_client;
 use crate::shard_interface::SessionShardInterface;
 use crate::{prometheus_metrics, XetFileInfo};
 
+lazy_static! {
+    pub static ref CONCURRENT_FILE_INGESTION_LIMITER: GlobalSemaphoreHandle =
```

Comment:
So static attributes are the thing that's bad for fork safety, but the global_semaphore_handle! is cheap to call on construction and returns the exact same one each time.